DEVINFO.md

```diff
@@ -0,0 +1,174 @@
+
+Therefore, making changes to the codebase is necessarily a review-intensive process.  No one wants bugs, but **no one can afford consensus bugs**.  This page describes how to make and review _non-consensus_ changes.  The process for consensus changes includes not only the entirety of this document, but also the [SIP process](https://github.com/stacksgov/sips/blob/main/sips/sip-000/sip-000-stacks-improvement-proposal-process.md).
+
+A good PR review sets both the submitter and reviewers up for success.  It minimizes the time required by both parties to get the code into an acceptable state, without sacrificing quality or safety.  Unlike most other software development practices, _safety_ is the primary concern.  A PR can and will be delayed or closed if there is any concern that it will lead to unintended consensus-breaking changes.
+
+This document describes some best practices on how to create and review PRs in this context.  The target audience is people who have commit access to this repository (reviewers), and people who open PRs (submitters).  This is a living document -- developers can and should document their own additional guidelines here.
+
+This document is formatted like a checklist.  Each paragraph is one goal or action item that the reviewer and/or submitter must complete.  The **key take-away** from each paragraph is bolded.
+
+## Reviewer Expectations
+
+The overall task of a reviewer is to create an **acceptance plan** for the submitter.  This is simply the list of things that the submitter _must_ do in order for the PR to be merged.  The acceptance plan should be coherent, cohesive, succinct, and complete enough that the reviewer will understand exactly what they need to do to make the PR worthy of merging, without further reviews.  The _lack of ambiguity_ is the most important trait of an acceptance plan.
+
+Reviewers should **strive to complete the review in one round**.  The reviewer should provide enough detail to the submitter that the submitter can make all of the requested changes without further supervision.  Whenever possible, the reviewer should provide all of these details publicly as comments, so that _other_ reviewers can vet them as well.  If a reviewer _cannot_ complete the review in one round due to its size and complexity, then the reviewer may request that the PR be simplified or broken into multiple PRs.
```

Comment:
I would add that reviews should use Github's "add comment as part of a review" feature to make the review atomic. Getting lots of notifications as each comment comes through "live" is not only annoying (maybe only I find this annoying?) but also increases the chances that a review ends up being multi-part.

Reply:
I'm ambivalent about this.  I just wait for the review comment arrival rate to slow down before opening the email thread, but that's just me. 

Maybe the submitter can ask for a preferred mode of receiving the review?

Reply:
I think we should tell reviewers to do this -- in addition to cutting down on notification pings, it also enforces review atomicity. I know I'm guilty of getting halfway through a review and then having to come back later. If I don't use the "pending comments" feature, it leaves the submitter not knowing whether or not the review was complete.

Reply:
This is specified in `CONTRIBUTING.md` now and I'll leave it there. If we want to remove this requirement later on then we can do that.